File: compiler/testData/objc/java/returnType/objcObjectIsCheck.kt
package test

import objc.OKPrinter
import objc.Printer
import objc.PrinterProvider

fun main(args: Array<String>) {
    for (i in 0..10) {
        val printer = PrinterProvider.getPrinter()
        if (printer is OKPrinter) {
            printer.print()
            return
        }
    }

    print("Fail")
}

File: compiler/testData/objc/java/returnType/objcObjectIsCheck.m
#import "objcObjectIsCheck.h"
#import <stdio.h>

@implementation Printer
- (void) print {
    printf("Generic printer");
}
@end

@implementation OKPrinter
- (void) print {
    printf("OK");
}
@end

@implementation FailPrinter
- (void) print {
    printf("Fail");
}
@end


@implementation PrinterProvider
+ (Printer *) getPrinter {
    static int times = 0;
    if (times++ < 5) {
        return [[FailPrinter alloc] init];
    } else {
        return [[OKPrinter alloc] init];
    }
}
@end
